Gravity/Gravity.Test.Unit/RdoExtensionsTests.cs

```diff
@@ -0,0 +1,192 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using Gravity.Base;
+using Gravity.Extensions;
+using Gravity.Test.TestClasses;
+using kCura.Relativity.Client.DTOs;
+using NUnit.Framework;
+using NUnit.Framework.Constraints;
+
+namespace Gravity.Test.Unit
+{
+	public class RdoExtensionsTests
+	{
+		[Test]
+		[TestCaseSource(nameof(ToHydratedDto_SimpleTypesFields_TestCases))]
+		public void ToHydratedDto_SimpleTypesFields(string fieldName, object value)
+		{
+			var property = typeof(GravityLevelOne).GetProperty(fieldName);
+			var propertyGuid = property.GetCustomAttribute<RelativityObjectFieldAttribute>().FieldGuid;
+
+			//make sure what you put into the RDO gets copied to the DTO
+			var dto = GetRdoWithField(propertyGuid, new FieldValue() { Value = value });
+
+			Assert.AreEqual(value, property.GetValue(dto));
+		}
+
+		public static IEnumerable<TestCaseData> ToHydratedDto_SimpleTypesFields_TestCases()
+		{
+			return new(string, object)[]
+			{
+				(nameof(GravityLevelOne.BoolField), true),
+				(nameof(GravityLevelOne.CurrencyField), 10.5m),
+				(nameof(GravityLevelOne.DateTimeField), new DateTime(2008, 1, 1)),
+				(nameof(GravityLevelOne.DecimalField), 10.5m),
+				(nameof(GravityLevelOne.LongTextField), "Test"),
+				(nameof(GravityLevelOne.FixedTextField), "Test"),
+				(nameof(GravityLevelOne.IntegerField), 1),
+				(nameof(GravityLevelOne.UserField), new User(1)),
+			}.Select(x => new TestCaseData(x.Item1, x.Item2).SetName("{m}(" + x.Item1 + ")"));
+		}
+
+		[Test]
+		[TestCaseSource(nameof(ToHydratedDto_TypeMismatch_TestCases))]
+		public void ToHydratedDto_TypeMismatch(string fieldName, object value)
+		{
+			Exception AssertThrowsAny(TestDelegate code)
+			{
+				return Assert.Throws<Exception>(() =>
+				{
+					try { code(); }
+					catch { throw new Exception(); }
+				});
+			}
+
+			var property = typeof(GravityLevelOne).GetProperty(fieldName);
+			var propertyGuid = property.GetFieldGuidValueFromAttribute();
+
+			AssertThrowsAny(() => GetRdoWithField(propertyGuid, new FieldValue() { Value = value }));
+		}
+
+		public static IEnumerable<TestCaseData> ToHydratedDto_TypeMismatch_TestCases()
+		{
+			return new[]
+			{
+				new TestCaseData(null, null)
+					.Ignore("Not worth investing in checking failure behavior before working on success behavior ")
+			};
+
+			return typeof(GravityLevelOne)
```

Comment:
This will never get hit right.  Are you just leaving for example.  Maybe comment out and explain.